--- Makefile
+++ Makefile
@@ -3,8 +3,8 @@
 # (C) 1997-2006 Nelson Murilo, Pangeia Informatica, AMS Foundation and others.
 #
 
-CC       = gcc
-CFLAGS	 = -DHAVE_LASTLOG_H
+CC       ?= gcc
+CFLAGS	 += -DHAVE_LASTLOG_H
 STATIC   = -static
 
 ###
@@ -29,48 +29,36 @@
 #CFLAGS	 =
 
 
-SRCS   = chklastlog.c chkwtmp.c ifpromisc.c chkproc.c chkdirs.c check_wtmpx.c strings.c
+SRCS   = chklastlog.c chkwtmp.c ifpromisc.c chkproc.c chkdirs.c strings.c
 
-OBJS   = chklastlog.o chkwtmp.o ifpromisc.o chkproc.o chkdirs.o check_wtmpx.o strings-static.o
+OBJS   = chklastlog.o chkwtmp.o ifpromisc.o chkproc.o chkdirs.o strings-static.o
 
 all:
 	@echo '*** stopping make sense ***'
 	@exec make sense
 
-sense: chklastlog chkwtmp ifpromisc chkproc chkdirs check_wtmpx strings-static chkutmp
+sense: chklastlog chkwtmp ifpromisc chkproc chkdirs strings-static chkutmp
 
 chklastlog:   chklastlog.c
-	${CC} ${CFLAGS} -o $@ chklastlog.c
-	@strip $@
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ chklastlog.c
 
 chkwtmp:   chkwtmp.c
-	${CC} ${CFLAGS} -o $@ chkwtmp.c
-	@strip $@
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ chkwtmp.c
 
 ifpromisc:   ifpromisc.c
 	${CC} ${CFLAGS} ${LDFLAGS}  -D_FILE_OFFSET_BITS=64 -o $@ ifpromisc.c
-	@strip $@
 
 chkproc:   chkproc.c
-	${CC} ${LDFLAGS} -o $@ chkproc.c
-	@strip $@
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ chkproc.c
 
 chkdirs:   chkdirs.c
-	${CC} ${LDFLAGS} -o $@ chkdirs.c
-	@strip $@
-
-check_wtmpx:   check_wtmpx.c
-	${CC} ${LDFLAGS} -o $@ check_wtmpx.c
-	@strip $@
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ chkdirs.c
 
 chkutmp:   chkutmp.c
-	${CC} ${LDFLAGS} -o $@ chkutmp.c
-	@strip $@
-
+	${CC} ${CFLAGS} ${LDFLAGS} -o $@ chkutmp.c
 
 strings-static:   strings.c
-	${CC} ${STATIC} ${LDFLAGS} -o $@ strings.c
-	@strip $@
+	${CC} ${STATIC} ${CFLAGS} ${LDFLAGS} -o $@ strings.c
 
 clean:
 	rm -f ${OBJS} core chklastlog chkwtmp ifpromisc chkproc chkdirs check_wtmpx strings-static chkutmp
