diff --exclude='*~' --exclude='.*' -I '$Id:' -urN chkrootkit-0.46.orig/chkproc.c chkrootkit-0.46/chkproc.c
--- chkrootkit-0.46.orig/chkproc.c	2005-10-27 08:29:26.000000000 -0400
+++ chkrootkit-0.46/chkproc.c	2005-10-27 08:30:42.000000000 -0400
@@ -43,6 +43,7 @@
 int main (){ return 0; }
 #else
 #include <stdio.h>
+#include <string.h>
 #include <errno.h>
 #include <sys/types.h>
 #include <dirent.h>
diff --exclude='*~' --exclude='.*' -I '$Id:' -urN chkrootkit-0.46.orig/chkwtmp.c chkrootkit-0.46/chkwtmp.c
--- chkrootkit-0.46.orig/chkwtmp.c	2005-10-27 08:29:26.000000000 -0400
+++ chkrootkit-0.46/chkwtmp.c	2005-10-27 08:30:45.000000000 -0400
@@ -21,6 +21,7 @@
 
 
 #include <stdio.h>
+#include <stdlib.h>
 #include <unistd.h>
 #include <string.h>
 #include <utmp.h>
