diff -Naur ../work/mc-4.1.40-pre9/lib/mc.menu mc-4.1.40-pre9/lib/mc.menu
--- ../work/mc-4.1.40-pre9/lib/mc.menu	2004-08-30 23:29:14.000000000 +0400
+++ mc-4.1.40-pre9/lib/mc.menu	2005-04-05 09:33:31.906924792 +0400
@@ -27,14 +27,23 @@
         info
 
 = t d
-3       Make a release of the current subdirectory
+3       Compress the current subdirectory (tar.gz)
+	Pwd=`basename "%d" /`
+	echo -n "Name of the compressed file (without extension) [$Pwd]: "
+	read tar
+        if [ "$tar"x = x ]; then tar="$Pwd"; fi
+	cd .. && \
+	tar cf - "$Pwd" | gzip -f9 > "$tar.tar.gz" && \
+	echo "../$tar.tar.gz created."
+
+4       Compress the current subdirectory (tar.bz2)
 	Pwd=`basename %d /`
-	echo -n "Name of the distribution file (without extension) [$Pwd]: "
+	echo -n "Name of the compressed file (without extension) [$Pwd]: "
 	read tar
-        if [ "$tar"x = x ]; then tar=$Pwd; fi
-	cd .. && tar chf - $Pwd | gzip -f9 > $tar.tar.gz
-	echo ../${tar}.tar.gz created.
-#       tar=%{Name of the distribution file (without extension)}
+        if [ "$tar"x = x ]; then tar="$Pwd"; fi
+	cd .. && \
+	tar cf - "$Pwd" | bzip2 -f > "$tar.tar.bz2" && \
+	echo "../$tar.tar.bz2 created."
 
 = f \.c$ & t r
 + f \.c$ & t r & ! t t
@@ -178,15 +187,27 @@
 	fi
 	echo "Please test the output file before deleting anything"
 
-=+ f \.tar\.gz$ | f \.tar\.z$ | f \.tgz$ | f \.tpz$ | f \.tar\.Z$ & t r
+=+ f \.tar\.gz$ | f \.tar\.z$ | f \.tgz$ | f \.tpz$ | f \.tar\.Z$| f \.tar\.bz2$ & t r
 x       Extract the contents of a compressed tar file
-	tar xzvf %f
+	unset EXT
+	case %f in 
+	    *.tar.bz2) EXT=tar_bz2;;
+	esac
+	if [ "$EXT" = "tar_bz2" ] 
+	then
+	  bunzip2 -c %f | tar xvf -
+	else
+	  gzip -dc %f | tar xvf -
+	fi
 
 = t r
 + ! t t
 y       Gzip or gunzip current file
         unset DECOMP
-	echo %f|egrep -q "\.gz$|\.z$|\.Z$" && DECOMP=-d
+	case %f in 
+	    *.gz) DECOMP=-d;;
+	    *.[zZ]) DECOMP=-d;;
+	esac
         gzip $DECOMP -v %f
 
 + t t
@@ -194,54 +215,126 @@
         for i in %t
         do
           unset DECOMP
-	  echo $i|egrep -q "\.gz$|\.z$|\.Z$" && DECOMP=-d
+	  case $i in 
+	    *.gz) DECOMP=-d;;
+	    *.[zZ]) DECOMP=-d;;
+	  esac
           gzip $DECOMP -v $i
         done
 
-+ f \.tar.gz$ | f \.tgz$ | f \.tpz$ | f \.tar.Z$ | f \.tar.z$ & t r & ! t t
++ ! t t
+b       Bzip2 or bunzip2 current file
+        unset DECOMP
+	case %f in 
+	    *.bz2) DECOMP=-d;;
+	esac
+        bzip2 $DECOMP -v %f
+
++ t t
+B       Bzip2 or bunzip2 tagged files
+        for i in %t
+        do
+          unset DECOMP
+	  case $i in 
+	    *.bz2) DECOMP=-d;;
+	  esac
+          bzip2 $DECOMP -v $i
+        done
+
++ f \.tar.gz$ | f \.tgz$ | f \.tpz$ | f \.tar.Z$ | f \.tar.z$ | f \.tar.bz2$ | f \.tar.F$ & t r & ! t t
 z       Extract compressed tar file to subdirectory
+	unset D
+	set gzip -cd
+	case %f in
+	  *.tar.gz) D="`basename %f .tar.gz`";;
+	  *.tgz)    D="`basename %f .tgz`";;
+	  *.tpz)    D="`basename %f .tpz`";;
+	  *.tar.Z)  D="`basename %f .tar.Z`";;
+	  *.tar.z)  D="`basename %f .tar.z`";;
+	  *.tar.bz2) D="`basename %f .tar.bz2`"; set bunzip2 -c ;;
+	  *.tar.F) D="`basename %f .tar.F`"; set freeze -dc;
+	esac
+	mkdir $D; cd $D && ($1 $2 ../%f | tar xvf -)
+
++ t t
+Z       Extract compressed tar files to subdirectories
+	for i in %t
+        do
+	  set gzip -dc
+          unset D
+	  case $i in
+	    *.tar.gz)  D="`basename $i .tar.gz`";;
+	    *.tgz)     D="`basename $i .tgz`";;
+	    *.tpz)     D="`basename $i .tpz`";;
+	    *.tar.Z)   D="`basename $i .tar.Z`";;
+	    *.tar.z)   D="`basename $i .tar.z`";;
+	    *.tar.F)   D="`basename $i .tar.F`"; set freeze -dc;;
+	    *.tar.bz2) D="`basename $i .tar.bz2`"; set bunzip2 -c;;
+          esac
+	  mkdir $D; (cd $D && $1 $2 ../$i | tar xvf -)
+        done
+
++ f \.gz$ | f \.tgz$ | f \.tpz$ | f \.Z$ | f \.z$ | f \.bz2$ & t r & ! t t
+c       Convert gz<->bz2, tar.gz<->tar.bz2 & tgz->tar.bz2
         unset D
-	echo %f|egrep -q "\.tar.gz$" && EXT=tar_gz
-	echo %f|egrep -q "\.tgz$"    && EXT=tgz
-	echo %f|egrep -q "\.tpz$"    && EXT=tpz
-	echo %f|egrep -q "\.tar.Z$"    && EXT=tar_Z
-	echo %f|egrep -q "\.tar.z$"    && EXT=tar_z
+	case %f in
+	    *.tgz)	EXT=tgz;;
+	    *.tpz)	EXT=tpz;;
+	    *.Z)	EXT=Z;;
+	    *.z)	EXT=z;;
+	    *.gz)	EXT=gz;;
+	    *.bz2)	EXT=bz2;;
+	esac
         case $EXT in
-          tar_gz) D="`basename %f .tar.gz`";;
-          tgz)    D="`basename %f .tgz`";;
-          tpz)    D="`basename %f .tpz`";;
-          tar_Z)  D="`basename %f .tar.Z`";;
-          tar_z)  D="`basename %f .tar.z`";;
+	  tgz|tpz)	D="`basename %f .$EXT`.tar";;
+	  gz|Z|z)	D="`basename %f .$EXT`";;
+	  bz2)		D="`basename %f .bz2`";;
         esac
-        mkdir $D; cd $D && tar xvzof ../%f
+        if [ "$EXT" = "bz2" ]; then
+	    bunzip2 -v %f ; gzip -f9 -v $D
+	else
+	    gunzip -v %f ; bzip2 -v $D
+	fi
 
 + t t
-Z       Extract compressed tar files to subdirectories
-	set %u
+C       Convert gz<->bz2, tar.gz<->tar.bz2 & tgz->tar.bz2
+	set %t
 	while [ -n "$1" ]
         do
           unset D
-	  echo $1|egrep -q "\.tar.gz$" && EXT=tar_gz
-	  echo $1|egrep -q "\.tgz$"    && EXT=tgz
-	  echo $1|egrep -q "\.tpz$"    && EXT=tpz
-	  echo $1|egrep -q "\.tar.Z$"    && EXT=tar_Z
-	  echo $1|egrep -q "\.tar.z$"    && EXT=tar_z
-          case $EXT in
-            tar_gz) D="`basename $1 .tar.gz`";;
-            tgz)    D="`basename $1 .tgz`";;
-            tpz)    D="`basename $1 .tpz`";;
-            tar_Z)  D="`basename $1 .tar.Z`";;
-            tar_z)  D="`basename $1 .tar.z`";;
-          esac
-          mkdir $D; (cd $D && tar xvzof ../$1)||echo tag $1 >>$MC_CONTROL_FILE
+	  case $1 in
+	    *.tgz)	EXT=tgz;;
+	    *.tpz)	EXT=tpz;;
+	    *.Z)	EXT=Z;;
+	    *.z)	EXT=z;;
+	    *.gz)	EXT=gz;;
+	    *.bz2)	EXT=bz2;;
+	  esac
+	  case $EXT in
+	      tgz)    D="`basename $1 .tgz`.tar";;
+              tpz)    D="`basename $1 .tpz`.tar";;
+	      gz|Z|z) D="`basename $1 .$EXT`";;
+	      bz2)    D="`basename $1 .bz2`";;
+	  esac
+	  if [ "$EXT" = "bz2" ]; then
+	    bunzip2 -v $1
+	    gzip -f9 -v $D
+	  else
+	    gunzip -v $1
+	    bzip2 -v $D
+	  fi
 	  shift
         done
 
++ x /usr/bin/open | x /usr/local/bin/open & x /bin/sh
+o       Open next a free console
+        open -s -- sh
+
 ##########################################################################
 # cdrtools section by Olegarch
 
 + t d
-B       Burn current dir [52x, JSS, koi8-r]
+J       Burn current dir [52x, JSS, koi8-r]
         COMMAND="mkisofs -a -f -jcharset koi8-r -J %d | cdrecord dev=0,0,0 speed=52 -v -data -"
 	echo Executing: ${COMMAND}
 	echo [CTRL+C to abort, ENTER to continue]

