diff -uNr vmmon-only.orig/linux/driver.c vmmon-only/linux/driver.c
--- vmmon-only.orig/linux/driver.c	2011-12-06 21:44:22.000000000 +0200
+++ vmmon-only/linux/driver.c	2011-12-06 21:47:44.000000000 +0200
@@ -179,7 +179,25 @@
 #endif
 };
 
-static struct file_operations vmuser_fops;
+/*
+ * Moved file operations initialize here because of incompatibilites
+ * with Gentoo hardened profile/hardend Linux 3.
+ */
+static struct file_operations vmuser_fops = {
+    .owner = THIS_MODULE,
+    .poll = LinuxDriverPoll,
+#ifdef HAVE_UNLOCKED_IOCTL
+    .unlocked_ioctl = LinuxDriver_UnlockedIoctl,
+#else
+    .ioctl = LinuxDriver_Ioctl,
+#endif
+#ifdef HAVE_COMPAT_IOCTL
+    .compat_ioctl = LinuxDriver_UnlockedIoctl,
+#endif
+    .open = LinuxDriver_Open,
+    .release = LinuxDriver_Close,
+    .mmap = LinuxDriverMmap
+};
 static struct timer_list tscTimer;
 
 /*
@@ -358,27 +376,6 @@
    spin_lock_init(&linuxState.pollListLock);
 #endif
 
-   /*
-    * Initialize the file_operations structure. Because this code is always
-    * compiled as a module, this is fine to do it here and not in a static
-    * initializer.
-    */
-
-   memset(&vmuser_fops, 0, sizeof vmuser_fops);
-   vmuser_fops.owner = THIS_MODULE;
-   vmuser_fops.poll = LinuxDriverPoll;
-#ifdef HAVE_UNLOCKED_IOCTL
-   vmuser_fops.unlocked_ioctl = LinuxDriver_UnlockedIoctl;
-#else
-   vmuser_fops.ioctl = LinuxDriver_Ioctl;
-#endif
-#ifdef HAVE_COMPAT_IOCTL
-   vmuser_fops.compat_ioctl = LinuxDriver_UnlockedIoctl;
-#endif
-   vmuser_fops.open = LinuxDriver_Open;
-   vmuser_fops.release = LinuxDriver_Close;
-   vmuser_fops.mmap = LinuxDriverMmap;
-
 #ifdef VMX86_DEVEL
    devel_init_module();
    linuxState.minor = 0;
