--- kernel/nv-linux.h.orig	2010-08-28 05:28:03.000000000 +0200
+++ kernel/nv-linux.h	2011-05-27 12:47:08.175727041 +0200
@@ -639,7 +639,7 @@
 #define NV_KMEM_CACHE_CREATE(kmem_cache, name, type)            \
     {                                                           \
         kmem_cache = kmem_cache_create(name, sizeof(type),      \
-                        0, 0, NULL);                            \
+                        0, SLAB_USERCOPY, NULL);                            \
     }
 #else
 #error "NV_KMEM_CACHE_CREATE_ARGUMENT_COUNT value unrecognized!"
