Patch by John Soros <johnny@9souldier.org>
--- lib/wmsetup.prev    2009-06-08 14:39:03.743529000 +0200
+++ lib/wmsetup 2009-06-08 14:41:08.446017443 +0200
@@ -27,6 +27,11 @@
        } >[2] /dev/null
 }

+### Added by gentoo, so acme uses the fonts distributed by acme-sac by default
+fn acme {
+       builtin acme -f /fonts/vera/Vera/unicode.14.font -F /fonts/vera/VeraMono/VeraMono.14.font $*
+}
+
 plumber >/chan/wmstdout >[2=1]
 menu Shell                     {wmrun wm/sh}
 menu Acme                      {wmrun acme}

