http://code.google.com/p/inferno-os/issues/detail?id=122
--- a/emu/port/inferno.c	2009-02-16 01:22:50.000000000 +0200
+++ b/emu/port/inferno.c	2008-10-23 00:04:07.000000000 +0300
@@ -760,10 +760,10 @@
 
 	fio = H2D(Sys_FileIO*, h);
 
-	c = cnewc(FioTread, movtmp, 16);
+	c = cnewc(FioTread, movtmp, 1024);
 	fio->read = c;
 
-	c = cnewc(FioTwrite, movtmp, 16);
+	c = cnewc(FioTwrite, movtmp, 1024);
 	fio->write = c;
 
 	f = fp;
