diff -uNr man.orig/1/sort man/1/sort
--- man.orig/1/sort	2012-12-03 01:03:27.257239571 +0200
+++ man/1/sort	2012-12-03 01:03:37.542239706 +0200
@@ -1,5 +1,5 @@
 .TH SORT 1
-.TH NAME
+.SH NAME
 sort \- sort file
 .SH SYNOPSIS
 .B sort
diff -uNr man.orig/3/tv man/3/tv
--- man.orig/3/tv	2012-12-03 01:03:27.258239571 +0200
+++ man/3/tv	2012-12-03 01:04:11.422240149 +0200
@@ -1,7 +1,7 @@
 .TH TV 3
 .SH NAME
 tv \- Hauppage TV device
-.SH NAME
+.SH SYNOPSIS
 .B
 bind -b #V /dev
 .PP
