--- appl/spree/mkfile.orig	2009-11-17 06:19:16.000000000 +0200
+++ appl/spree/mkfile	2009-11-17 06:19:47.000000000 +0200
@@ -54,7 +54,7 @@
 	cp $prereq $target
 
 %.dis:	%.b
-	limbo -gw -I lib -o $stem.dis $stem.b
+	limbo -gw -I lib -I $ROOT/module -o $stem.dis $stem.b
 
 $ENGINES $MAIN $LIB: spree.m gather.m lib/cardlib.m lib/allow.m lib/objstore.m
 $ENGINES $MAIN $CLIENTS $LIB: ${MODULES:%=$ROOT/module/%}
