diff -uNr go-9999.orig/src/make.bash go-9999/src/make.bash
--- go-9999.orig/src/make.bash	2012-02-16 18:12:29.022073650 +0200
+++ go-9999/src/make.bash	2012-02-16 18:14:41.343075384 +0200
@@ -82,6 +82,37 @@
 ./cmd/dist/dist bootstrap $buildall -v # builds go_bootstrap
 # Delay move of dist tool to now, because bootstrap may clear tool directory.
 mv cmd/dist/dist "$GOTOOLDIR"/dist
+if [ -f "$GOTOOLDIR"/8l ]; then
+    mv "$GOTOOLDIR"/8l{,.orig}
+    cat <<'EOF' >"$GOTOOLDIR"/8l
+#!/bin/bash
+BIN="8.out"
+for (( i = 1 ; i <= $# ; i++ )); do
+    if [ "x${!i}" = "x-o" ]; then
+        i=$(( i + 1 ))
+        BIN="${!i}"
+    fi
+done
+$(dirname $0)/8l.orig "$@" && paxctl -cm "$BIN" &>/dev/null
+EOF
+    chmod +x "$GOTOOLDIR"/8l
+fi
+if [ -f "$GOTOOLDIR"/6l ]; then
+    mv "$GOTOOLDIR"/6l{,.orig}
+    cat <<'EOF' >"$GOTOOLDIR"/6l
+#!/bin/bash
+BIN="6.out"
+for (( i = 1 ; i <= $# ; i++ )); do
+    if [ "x${!i}" = "x-o" ]; then
+        i=$(( i + 1 ))
+        BIN="${!i}"
+    fi
+done
+$(dirname $0)/6l.orig "$@" && paxctl -cm "$BIN" &>/dev/null
+EOF
+    chmod +x "$GOTOOLDIR"/6l
+fi
+paxctl -cm "$GOTOOLDIR"/go_bootstrap &>/dev/null
 "$GOTOOLDIR"/go_bootstrap clean -i std
 echo
 
