diff -uNr a/cue2tracks b/cue2tracks
--- a/cue2tracks	2013-07-06 13:50:36.000000000 +0300
+++ b/cue2tracks	2013-09-14 22:26:24.728355048 +0300
@@ -3,6 +3,9 @@
 PATH=${PATH}:/bin:/usr/bin:/usr/local/bin
 export PATH=$PATH
 
+# set CPU_NUMBER to the number of your CPUs. This variable is used by flaccl.
+CPU_NUMBER=$(grep ^processor /proc/cpuinfo | wc -l)
+
 # --== ERRORS ==--
 
 chk_error=0
@@ -28,6 +31,7 @@
 
 E_CODEC_FLAC_NOT_FOUND=15
 E_CODEC_FLAKE_NOT_FOUND=16
+E_CODEC_FLACCL_NOT_FOUND=66
 E_CODEC_APE_NOT_FOUND=17
 E_CODEC_WVp_NOT_FOUND=18
 E_CODEC_WVu_NOT_FOUND=19
@@ -98,6 +102,7 @@
 
 codec_FLAC=""
 codec_FLAKE=""
+codec_FLACCL=""
 codec_APE=""
 codec_WVp=""
 codec_WVu=""
@@ -133,6 +138,7 @@
 
 codec_FLAC_needed=0
 codec_FLAKE_needed=0
+codec_FLACCL_needed=0
 codec_APE_needed=0
 codec_WVp_needed=0
 codec_WVu_needed=0
@@ -277,6 +283,7 @@
 \t\twav   : no encode, raw sound wave,\n\
 \t\tflac  : (default) Free Lossless Audio Codec,\n\
 \t\tflake : FLAC realisation via FFmpeg (beta),\n\
+\t\tflaccl: FLAC realisation using GPU acceleration,\n\
 \t\tape   : Monkey's Audio Codec,\n\
 \t\twv    : WavPack,\n\
 \t\tofr   : OptimFrog,\n\
@@ -403,6 +410,16 @@
 			[ ${outCodecLevel} -eq 2 ] && outCodecParam=" flake -4 - %f"
 			codec_OUT="flake"
 		;;
+		"flaccl" )
+			codec_FLACCL_needed=1
+			if [ ${putTags} -eq 1 ]; then tag_FLAC_needed=1; fi
+			outExt="flac"
+			toolCodec="${outExt}"
+			outCodecParam=" flaccl --cpu-threads ${CPU_NUMBER} -o %f -"
+			[ ${outCodecLevel} -eq 1 ] && outCodecParam=" flaccl -1 --cpu-threads ${CPU_NUMBER} -o %f -"
+			[ ${outCodecLevel} -eq 2 ] && outCodecParam=" flaccl -4 --cpu-threads ${CPU_NUMBER} -o %f -"
+			codec_OUT="flaccl"
+		;;
 		"tak" )
 			codec_TAK_needed=1
 			if [ ${putTags} -eq 1 ]; then tag_APE_needed=1; fi
@@ -695,6 +712,13 @@
 		[ -n "${codec_FLAKE}" ] || return ${E_CODEC_FLAKE_NOT_FOUND}
 	fi
 
+	if [ ${codec_FLACCL_needed} -eq 1 ]
+	then
+		checktool flaccl flaccl
+		codec_FLACCL=`gettool flaccl`
+		[ -n "${codec_FLACCL}" ] || return ${E_CODEC_FLACCL_NOT_FOUND}
+	fi
+
 	if [ ${codec_OFR_needed} -eq 1 ]
 	then
 		checktool ofr optimfrog
@@ -1227,7 +1251,7 @@
 # function for track taging
 tag_track() {
 	case "${outCodec}" in
-		"flac" | "flake" )
+		"flac" | "flake" | "flaccl" )
 			if [ ${onlyTest} -eq 0 ]
 			then
 				[ -n "${tags_ALBUM_DISCID}" ] && ( tag_flac_track "DISCID" "${tags_ALBUM_DISCID}" || return ${E_CANT_TAG_FLAC} )
