diff -uNr cue2tracks-0.2.11.orig/cue2tracks cue2tracks-0.2.11/cue2tracks
--- cue2tracks-0.2.11.orig/cue2tracks	2009-06-21 15:13:30.000000000 +0300
+++ cue2tracks-0.2.11/cue2tracks	2010-11-07 19:31:05.000000000 +0200
@@ -764,6 +764,7 @@
 		bnc=`basename "${cueFile}" .cue`
 		ncn="${tempDirName}/${bnc}.line.cue"
 		cp "${cueFile}" "${ncn}"
+		chmod +w "${ncn}"
 		echo -e "\n" >> "${ncn}"
 		[ $? -ne 0 ] && return ${E_CANT_FIX_CUE}
 		cueFile="${ncn}"
