--- /usr/inferno/appl/cmd/ndb/registry.b	2009-02-17 00:05:18.000000000 +0200
+++ appl/cmd/ndb/registry.b	2009-02-22 05:41:11.000000000 +0200
@@ -50,6 +50,7 @@
 Shift:	con 4;
 Mask:	con 2r1111;
 
+Einterrupted: con "interrupted";
 Egreg: con "buggy program!";
 Maxreplyidle: con 3;
 
@@ -92,6 +93,7 @@
 	remove:	fn(e: self ref Event);
 	queue:	fn(e: self ref Event, m: ref Tmsg.Read): string;
 	post:		fn(vers: int);
+	flush:	fn(tag: int);
 };
 
 filters: list of ref Filter;
@@ -214,6 +216,9 @@
 			err = remove(fid);
 			if(err == nil)
 				srv.reply(ref Rmsg.Remove(m.tag));
+		Flush =>
+			Event.flush(m.oldtag);
+			srv.default(gm);
 		* =>
 			srv.default(gm);
 		}
@@ -696,6 +701,18 @@
 	}
 }
 
+Event.flush(tag: int)
+{
+	for(l := events; l != nil; l = tl l){
+		e := hd l;
+		if(e.m != nil && e.m.tag == tag){
+			srv.reply(ref Rmsg.Error(e.m.tag, Einterrupted));
+			e.m = nil;
+			break;
+		}
+	}
+}
+
 dbload(db: ref Db)
 {
 	ptr: ref Attrdb->Dbptr;
