diff -uNr unlock-all-v110.orig/src/Unlocker.cpp unlock-all-v110/src/Unlocker.cpp
--- unlock-all-v110.orig/src/Unlocker.cpp	2012-03-20 12:29:34.000000000 +0200
+++ unlock-all-v110/src/Unlocker.cpp	2012-06-11 17:03:12.145888369 +0300
@@ -78,7 +78,7 @@
 char const vmx_debug[] = "vmware-vmx-debug";
 char const vmx_stats[] = "vmware-vmx-stats";
 char const vmwarebase[] = "libvmwarebase.so.0";
-char const install_path[] = "/usr/lib/vmware/";
+char const install_path[] = "/opt/vmware/lib/vmware/";
 char const vmx_path[] = "bin/";
 char const vmwarebase_path[] = "lib/libvmwarebase.so.0/";
 #endif /* __ESXi__ || __linux__ */
